plugins/outputs/stackdriver/counter_cache.go

```diff
@@ -0,0 +1,96 @@
+package stackdriver
+
+import (
+	"path"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+
+	monpb "google.golang.org/genproto/googleapis/monitoring/v3"
+	tspb "google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type CounterCache struct {
+	sync.RWMutex
+	cache map[string]*CounterCacheEntry
+	log   telegraf.Logger
+}
+
+type CounterCacheEntry struct {
+	LastValue *monpb.TypedValue
+	StartTime *tspb.Timestamp
+}
+
+func (cce *CounterCacheEntry) Reset(ts *tspb.Timestamp) {
+	// always backdate a reset by -1ms, otherwise stackdriver's API will hate us
+	cce.StartTime = tspb.New(ts.AsTime().Add(time.Millisecond * -1))
+}
```

Comment:
I think none of the functions nor the structs need to be exported as they live in the same `stackdriver` package.

Reply:
👍